stablehlo/tests/interpret_sine.mlir

```diff
@@ -0,0 +1,111 @@
+// RUN: stablehlo-interpreter --interpret -split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: Evaluated results of function: sine_op_test_bf16
+func.func @sine_op_test_bf16() -> tensor<12xbf16> {
+  %0 = stablehlo.constant dense<[0.0, -0.0, 1.0, 0.125, 0.1, 3.140630, 0x7F80, 0xFF80, 0x7FFF, 0xFFFF, 0x0001, 0x8001]> : tensor<12xbf16>
```

Comment:
missing +/- Inf?
Keep one representation of NaN and denormals? 
here and below.

Reply:
The hex values in each test case currently are: +inf, -inf, +NaN, -NaN, +Denormal (smallest), -Denormal (largest). Since we are not differentiating the difference between +/- NaN, I will remove one. I will also create a separate issue to revisit and explore which tests are worth exploring (to add necessary tests and remove unnecessary tests).